core/frontend/src/IModelApp.ts

```diff
@@ -298,11 +298,17 @@ export class IModelApp {
   public static get publicPath() { return this._publicPath; }
 
   /** The [[FormatsProvider]] for this session.
-   * @param provider The provider to use for formatting quantities. If not provided, the provider will be cleared.
+   * @param provider The provider to use for formatting quantities.
    * @beta
    */
   public static get formatsProvider(): FormatsProvider { return this._formatsProvider; }
-  public static set formatsProvider(provider: FormatsProvider) { this._formatsProvider = provider }
+  public static set formatsProvider(provider: FormatsProvider) {
+    this._formatsProvider.formatsProvider = provider;
+    this._formatsProvider.formatsProvider.onFormatsChanged.addListener((args: FormatsChangedArgs) => {
```

Comment:
why not do this in `FormatsProviderManager.formatsProvider` setter?

Reply:
I was 50/50 on this yeah. I've moved the events logic into FormatsProviderManager